android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt

```diff
@@ -0,0 +1,95 @@
+package com.onegini.mobile.sdk.reactnative
+
+import android.net.Uri
+import com.facebook.react.bridge.JavaOnlyMap
+import com.facebook.react.bridge.Promise
+import com.onegini.mobile.sdk.android.handlers.OneginiAppToWebSingleSignOnHandler
+import com.onegini.mobile.sdk.android.handlers.error.OneginiAppToWebSingleSignOnError
+import com.onegini.mobile.sdk.android.model.OneginiAppToWebSingleSignOn
+import com.onegini.mobile.sdk.reactnative.clean.use_cases.StartSingleSignOnUseCase
+import com.onegini.mobile.sdk.reactnative.facade.UriFacade
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Answers
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+@RunWith(MockitoJUnitRunner::class)
+
+class StartSingleSignOnUseCaseTests {
+    @get:Rule
+    val reactArgumentsTestRule = ReactArgumentsTestRule()
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private lateinit var oneginiSdk: OneginiSDK
+
+    @Mock
+    private lateinit var uriFacade: UriFacade
+
+    // We need to deep stub here to mock a uri object's .toString() as we cant pass in a uriFacade
```

Comment:
as we cant pass `it into` uriFacade
isn't it better? 🤔

Reply:
It actually means something else so I have adjusted it to hopefully make it more understandable.